src/lib/y2storage/encryption_processes/swap.rb

```diff
@@ -17,60 +17,169 @@
+    #
+    # Swap based methods use plain encryption technology to encrypt the swap device with a new password
+    # generated at boot time. Generally, the new password is generated based on /dev/urandom data (see
+    # {RandomSwap}), but IBM offers other mechanisms for z Systems (see {ProtectedSwap} and {SecureSwap}
+    # processes).
     class Swap < Base
-      URANDOM = "/dev/urandom".freeze
-      SWAP = "swap".freeze
+      # @see Base.available?
+      def self.available?
+        File.exist?(key_file)
+      end
 
-      private_constant :URANDOM
-      private_constant :SWAP
+      # @see Base.only_for_swap?
+      def self.only_for_swap?
+        true
+      end
 
-      # Whether the process was used for the given encryption device
-      #
-      # @param encryption [Y2Storage::Encryption] the encryption device to check
-      # @return [Boolean] true when "swap" is found in the crypt options; false otherwise
+      # @see Base.used_for?
       def self.used_for?(encryption)
-        encryption.crypt_options.any? { |opt| opt.downcase == SWAP }
+        used?(encryption.key_file, encryption.crypt_options)
       end
 
-      # @see EncryptionProcesses::Base#only_for_swap?
-      def self.only_for_swap?
-        true
+      # @see Base.used_for_crypttab?
+      def self.used_for_crypttab?(entry)
+        used?(entry.password, entry.crypt_options)
       end
 
-      # @see EncryptionProcesses::Base#create_device
+      # @see Base#create_device
       def create_device(blk_device, dm_name)
         enc = super
         enc.key_file = key_file
         enc.crypt_options = crypt_options + enc.crypt_options
         enc
       end
 
-      private
-
-      # @see EncryptionProcesses::Base#encryption_type
+      # @see Base#encryption_type
       def encryption_type
         EncryptionType::PLAIN
       end
 
-      # Returns the options used for the encryption
-      #
-      # @return [Array<String>]
-      def crypt_options
-        [SWAP]
+      private
+
+      SWAP_OPTION = "swap".freeze
+      private_constant :SWAP_OPTION
+
+      class << self
+        private
+
+        # Encryption key file
+        #
+        # Each Swap process could use a different key file.
+        #
+        # @raise [RuntimeError] if no key file has been defined.
+        #
+        # @return [String]
+        def key_file
+          raise "No key file indicated!"
+        end
+
+        # Encryption cipher
+        #
+        # Each Swap process could use a different cipher.
+        #
+        # @return [String, nil] nil if no specific cipher
+        def cipher
+          nil
+        end
+
+        # Size of the encryption key
+        #
+        # @return [String, nil] nil if no specific size
+        def key_size
+          nil
+        end
+
+        # Encryption options to add to the encryption device
+        #
+        # @return [Array<String>]
+        def crypt_options
+          options = [swap_option]
```

Comment:
NP: since those _*\_option_ methods are actually returning `nil` when no specific value available, I think that simply 

```ruby
[swap_option, cipher_option, key_size_option].compact
```
should be enough.

But probably it's just a matter of taste :)